File: src/ServerPulse.Frontend/src/app/modules/shared/domain/dtos/statistics/response/statistics/serverLoadStatisticsResponse.ts
import { BaseStatisticsResponse, convertBaseResponse, convertToLoadMethodStatisticsResponse, convertToServerLoadResponse, LoadEventResponse, LoadMethodStatisticsResponse } from "../../../../../index";

export interface ServerLoadStatisticsResponse extends BaseStatisticsResponse {
    amountOfEvents: number;
    lastEvent: LoadEventResponse | null;
    loadMethodStatistics: LoadMethodStatisticsResponse | null;
}

export function convertToServerLoadStatisticsResponse(data: any): ServerLoadStatisticsResponse {
    return {
        ...convertBaseResponse(data),
        amountOfEvents: data?.AmountOfEvents ?? 0,
        lastEvent: convertToServerLoadResponse(data?.LastEvent),
        loadMethodStatistics: convertToLoadMethodStatisticsResponse(data?.LoadMethodStatistics)
    };
}

File: src/ServerPulse.Backend/AnalyzerApi.Infrastructure/Dtos/Responses/Statistics/ServerLoadStatisticsResponse.cs
using AnalyzerApi.Infrastructure.Dtos.Responses.Events;

namespace AnalyzerApi.Infrastructure.Dtos.Responses.Statistics
{
    public class ServerLoadStatisticsResponse : BaseStatisticsResponse
    {
        public int AmountOfEvents { get; set; }
        public LoadEventResponse? LastEvent { get; set; }
        public LoadMethodStatisticsResponse? LoadMethodStatistics { get; set; }
    }
}